contributors/design-proposals/node-allocatable.md

```diff
@@ -89,61 +73,206 @@ The flag will be specified as a serialized `ResourceList`, with resources define
+    * If a pod has a `RestartPolicy=Never`, then mark the pod as `Failed` and terminate its workload.
+    * All other pods that are not parented by new Allocatable top level cgroup will be restarted.
+  * Users intending to turn off this feature can set this flag to `""`.
+  * `--kube-reserved` and `--system-reserved` flags are expected to be set by users for this flag to have any meaningful effect on node stability.
+  * By including `kube-reserved` or `system-reserved` in this flag's value, and by specifying the following two flags, Kubelet will attempt to enforce the reservations specified via `--kube-reserved` & `system-reserved` respectively.
+
+2. `--kube-reserved-cgroup=<absolute path to a cgroup>`
+   * This flag helps kubelet identify the control group managing all kube components like Kubelet & container runtime that fall under the `KubeReserved` reservation.
+
+3. `--system-reserved-cgroup=<absolute path to a cgroup>`
+   * This flag helps kubelet identify the control group managing all OS specific system daemons that fall under the `SystemReserved` reservation.
+
+#### Rollout details
+
+This phase is expected to improve Kubernetes node stability. However it requires users to specify non-default values for `--kube-reserved` & `--system-reserved` flags though.
+
+The rollout of this phase has been long due and hence we are attempting to include it in v1.6
+
+Since `KubeReserved` and `SystemReserved` continue to have `""` as defaults, the node's `Allocatable` does not change automatically.
+Since this phase requires node drains (or pod restarts/terminations), it is considered disruptive to users.
+
+To rollback this phase, set `--enforce-node-allocatable` flag to `""`.
+
+This phase might cause the following symptoms to occur if `--kube-reserved` and/or `--system-reserved` flags are also specified.
+
+1. OOM kills of containers and/or evictions of pods. This can happen primarily to `Burstable` and `BestEffort` pods since they can no longer use up all the resource available on the node.
+
+##### Proposed Timeline
+
+02/14/2017 - Discuss the rollout plan in sig-node meeting
+02/15/2017 - Flip the switch to enable pod level cgroups by default
+enable existing experimental behavior by default
+02/21/2017 - Assess impacts based on enablement
+02/27/2017 - Kubernetes Feature complete (i.e. code freeze)
+03/01/2017 - Send an announcement to kubernetes-dev@ about this rollout along with rollback options and potential issues. Recommend users to set kube and system reserved.
+03/22/2017 - Kubernetes 1.6 release
+
+### Phase 3 - Metrics & support for Storage
+
+*Status*: Targetted for v1.7
+
+In this phase, Kubelet will expose usage metrics for `KubeReserved`, `SystemReserved` and `Allocatable` top level cgroups via Summary metrics API.
+`Storage` will also be introduced as a reservable resource in this phase.
+Support for evictions based on Allocatable will be introduced in this phase.
```

Comment:
> it seems like memory pressure not being reported back to the scheduler for allocatable is a step back.

That's true. We need to prevent new pods from being scheduled. But look below.

> Thinking about this more, users could set their eviction thresholds higher to cover system+kube reserved in the interim.

Let's assume the following:
1. user sets their eviction thresholds to say `85%` 
2. they enforce `SystemReserved` & `KubeReserved` which are set to `5%` each.
3. System and kube components use upto capacity.
4. Hence allocatable is `90%`.

Pods cannot use more than 75% of the memory. If they use more than that, they will be evicted anyways.

Given that do we have to do anything extra? Am I missing something?

I feel eviction thresholds limit the available capacity on the nodes for pods. If a node only runs Guaranteed pods, today, they cannot use upto `Allocatable` if user space evictions are turned on and that worries me.